src/aspnetcore/eng/Versions.props

```diff
@@ -203,6 +203,21 @@
     -->
     <MicrosoftNETCoreAppRuntimeVersion>$(MicrosoftNETCoreAppRuntimewinx64Version)</MicrosoftNETCoreAppRuntimeVersion>
   </PropertyGroup>
+  <!-- The maintenance-packages dependency versions need to be conditionally selected: https://github.com/dotnet/sdk/issues/45155 -->
+  <PropertyGroup Label="Dependencies from dotnet/maintenance-packages (build-source-only)" Condition="'$(DotNetBuildSourceOnly)' == 'true'">
+    <MicrosoftBCLHashCodeVersion>6.0.0</MicrosoftBCLHashCodeVersion>
+    <SystemBuffersVersion>4.6.0</SystemBuffersVersion>
+    <SystemThreadingTasksExtensionsVersion>4.6.0</SystemThreadingTasksExtensionsVersion>
+    <SystemValueTupleVersion>4.6.0</SystemValueTupleVersion>
```

Comment:
@ericstj from your comment: https://github.com/dotnet/dotnet/pull/133#issuecomment-2669729688

Already updated it here, but conditioned to source build.

I do see a bunch of csprojs in aspnet core that are hardcoding System.ValueTuple, instead of using the property value.

- eng/Baseline.Designer.props
- src/submodules/MessagePack-CSharp/sandbox/...
  - TestData.InvalidProject.csproj
  - TestData.InvalidSyntax.csproject
  - TestData2.csproj

Reply:
Those projects aren't the ones casing the failure.

Why is it OK to condition it on source build if you haven't conditioned it on source build in runtime?  Runtime has the newer reference, which comes in transitively then ASPNET project is downgrading it.